apps/browser/qml/pages/components/Overlay.qml

```diff
@@ -70,6 +70,11 @@ Shared.Background {
         overlayAnimator.showChrome()
     }
 
+    Connections {
+        target: webView.tabModel
+        onCountChanged: window.setBrowserCover(webView.tabModel)
```

Comment:
I'd add "!WebUtils.firstUseDone" condition to the no tabs cover code path of the setBrowserCover so that when starting browser for the very first time we'd also have no tabs cover visible. You can simulate first time usage by closing all tabs and then removing .local/share/org.sailfishos/browser/.firstUseDone. When/if browser is started for the first with a url, the firstUseDone will be touched in place after toolbar hits the bottom state for the very first time. That may imply that you need to add to the "target: WebUtils" connection (BrowserPage line 374) onFirstUseDoneChanged signal handler as well.

I'd keep ```window.setBrowserCover(webView.tabModel)``` in the BrowserPage.qml Component.onCompleted so that initial status will be updated. Further, there's already tabModel connection in the BrowserPage approximately line 207.